src/Core/Content/Seo/SeoUrlPlaceholderHandler.php

```diff
@@ -90,43 +87,55 @@ private function createSeoMapping(SalesChannelContext $context, array $mapping):
             return [];
         }
 
-        $query = new QueryBuilder($this->connection);
-        $query->setTitle('seo_url::replacement');
-        $query->addSelect(['seo_path_info', 'path_info', 'sales_channel_id']);
+        $subqueryParts = '';
+        $params = [];
 
-        $query->from('seo_url');
-        $query->andWhere('seo_url.is_canonical = 1');
-        $query->andWhere('seo_url.path_info IN (:pathInfo)');
-        $query->andWhere('seo_url.language_id = :languageId');
-        $query->andWhere('seo_url.sales_channel_id = :salesChannelId OR seo_url.sales_channel_id IS NULL');
-        $query->andWhere('seo_url.is_deleted = 0');
-        $query->setParameter('pathInfo', $mapping, ArrayParameterType::STRING);
-        $query->setParameter('languageId', Uuid::fromHexToBytes($context->getContext()->getLanguageId()));
-        $query->setParameter('salesChannelId', Uuid::fromHexToBytes($context->getSalesChannelId()));
-
-        $seoUrls = $query->executeQuery()->fetchAllAssociative();
+        foreach ($mapping as $key => $value) {
+            $subqueryParts .= '(SELECT id FROM seo_url WHERE path_info = ? AND language_id = 0x' . $context->getLanguageId() . ' AND is_canonical = 1 AND is_deleted = 0 AND sales_channel_id = 0x' . $context->getSalesChannelId() . ' LIMIT 1) UNION ALL ';
```

Comment:
Could you use a named parameter instead and use Uuid::fromFromHexToBytes instead of string concat for hex ids?

Reply:
I don't think we can use the named parameters for languageId & salesChannelId as we can't mix named & unnamed parameters.

Reply:
Could you change path_info to the named argument as well? then we can use the binding named parameter instead of concatting the sql query which is not recommended.

Reply:
We cannot use named parameters here because we cannot use a name multiple times. We would have to create a unique name for each query and I don't think that effort would be worth it.